tools/shared/src/main/scala/org/scalajs/core/tools/javascript/ScalaJSClassEmitter.scala

```diff
@@ -298,9 +361,9 @@ final class ScalaJSClassEmitter(
     require(tree.kind.isJSClass)
 
     tree.exportedMembers.map(_.tree) collectFirst {
-      case MethodDef(false, StringLiteral("constructor"), params, _, body) =>
+      case methodDef: MethodDef if !methodDef.static && methodDef.name == StringLiteral("constructor") =>
```

Comment:
Or better, keep the extracting pattern, and use the captured fields (since you don't need `methodDef` as such, do you?)